pyannote/audio/applications/multilabel.py

```diff
@@ -0,0 +1,584 @@
+         keep_missing: False    # Raise an error if one of the input label is not found in the mapping.
+         mapping:
+            "BRO": "CHI"
+            "MOT": "FEM"
+            "FAT": "MAL"
+    ...................................................................
+
+"train" mode: 
+    This will create the following directory that contains the pre-trained
+    neural network weights after each epoch: 
+
+        <experiment_dir>/train/<database.task.protocol>.<subset>
+
+    This means that the network was trained on the <subset> subset of the
+    <database.task.protocol> protocol. By default, <subset> is "train".
+    This directory is called <train_dir> in the subsequent "validate" mode.
+
+    A bunch of values (loss, learning rate, ...) are sent to and can be
+    visualized with tensorboard with the following command: 
+
+        $ tensorboard --logdir=<experiment_dir>
+
+"validate" mode: 
+    Use the "validate" mode to run validation in parallel to training.
+    "validate" mode will watch the <train_dir> directory, and run validation
+    experiments every time a new epoch has ended. This will create the
+    following directory that contains validation results: 
+
+        <train_dir>/validate/<database.task.protocol>.<subset>
+
+    You can run multiple "validate" in parallel (e.g. for every subset,
+    protocol, task, or database).
+
+    In practice, for each epoch, "validate" mode will look for the peak
+    detection threshold that maximizes speech turn coverage, under the
+    constraint that purity must be greater than the value provided by the
+    "--purity" option. Both values (best threshold and corresponding coverage)
+    are sent to tensorboard.
+
+"apply" mode
+    Use the "apply" mode to extract segmentation raw scores.
+    Resulting files can then be used in the following way: 
+
+    >>> from pyannote.audio.features import Precomputed
+    >>> precomputed = Precomputed('<output_dir>')
+
+    >>> from pyannote.database import get_protocol
+    >>> protocol = get_protocol('<database.task.protocol>')
+    >>> first_test_file = next(protocol.test())
+
+    >>> from pyannote.audio.signal import Peak
+    >>> peak_detection = Peak()
+
+    >>> raw_scores = precomputed(first_test_file)
+    >>> homogeneous_segments = peak_detection.apply(raw_scores, dimension=1)
+"""
+
+from os.path import dirname, basename
+from tqdm import tqdm
+from functools import partial
+from pathlib import Path
+import torch
+import numpy as np
+import scipy.optimize
+from docopt import docopt
+import multiprocessing as mp
+from .base_labeling import BaseLabeling
+
+from pyannote.database import get_annotated, get_protocol, FileFinder
+
+from pyannote.metrics.detection import DetectionErrorRate, DetectionRecall, DetectionPrecision
+
+from pyannote.audio.labeling.extraction import SequenceLabeling
+
+from pyannote.audio.pipeline import SpeakerActivityDetection \
+                             as SpeakerActivityDetectionPipeline
```

Comment:
Wouldn't it make more sense to rename the `SpeakerActivityDetection` pipeline as `MultiLabelClassification` or something like that?

Reply:
Having read the rest of the pull request, I don't think this new pipeline is needed.
It is actually very similar to the `SpeechActivityDetection` pipeline which should be used/updated instead.